sklearn/svm/_classes.py

```diff
@@ -1140,6 +1140,45 @@ def __init__(
             random_state=random_state,
         )
 
+    def fit(self, X, y, sample_weight=None):
+        """
+        Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        y : array-like of shape (n_samples,)
+            Target values (class labels).
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Per-sample weights. If None, then each sample is given equal weight.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+
+        if sample_weight is None and self.class_weight is not None:
```

Comment:
We should still do that if `sample_weight is not None` but instead we should multiply with the user provided sample weight.